### PR TITLE
backend: fix load partition table with local backend

### LINKED ISSUE #401: local backend miss a lot of index kv with partition table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
use lightning with local backend to load partition table data, checksum fails with lost a lot of kvs.

The data kv count is right, but index are much fewer.

The root cause of this bug is this method: 
https://github.com/pingcap/tidb-lightning/blob/12fe8b151119becc4cac0501806504e720e4e61d/lightning/backend/local.go#L618-L622

Since each partition is a separate physic table

And after use `admin repiar index`, table checksum matches local checksum.

2. What did you expect to see?

3. What did you see instead?

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):
        ```
        v4.0.5
        ```
        
    - TiKV-Importer version (run `tikv-importer -V`)

        ```
        (paste TiKV-Importer version here)
        ```
        
    - TiKV version (run `tikv-server -V`):

        ```
        (paste TiKV version here)
        ```

    - TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - Other interesting information (system version, hardware config, etc):

        

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `inventory.ini` if deployed by Ansible

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for TiDB-Lightning if possible

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
close #401 

### What is changed and how it works?
Try to split index range for each physical table in the index engine instead of only process the first one.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

File: lightning/backend/local.go [hunk 1]
```diff
@@ -614,60 +614,44 @@ func (local *local) readAndSplitIntoRange(engineFile *LocalFile, engineUUID uuid
...
-			iter.SeekGE(k)
-			// get first key of index i
-			startKeyOfIndex := append([]byte{}, iter.Key()...)
-
-			k = tablecodec.EncodeTableIndexPrefix(tableID, i+1)
-			// get last key of index i
-			iter.SeekLT(k)
 
-			lastKeyOfIndex := append([]byte{}, iter.Key()...)
+		// for partition table, there will be multi physical table and physical each table contains multiple index
```
Comment (kennytm) [inline lightning/backend/local.go:624]:
```suggestion
		// for partitioned table, there will be multiple physical tables and each physical table contains multiple indices
```

**Review by kennytm — COMMENTED** [review]
Rest LGTM

**Review by 3pointer — APPROVED** [review]
lgtm

**Pull request merged**